Vungle Sample App Swift/Vungle Sample App Swift/ViewController.swift

```diff
@@ -226,30 +156,96 @@ class ViewController: UIViewController, VungleSDKDelegate {
     
     private func showAdForPlacement03 () {
         // Play a Vungle ad (with options). Dictionary to set custom ad options.
-        let options: NSDictionary = NSDictionary(dictionary: [VunglePlayAdOptionKeyUser: "test_user_id",
-                                                              VunglePlayAdOptionKeyIncentivizedAlertBodyText: "If the video isn't completed you won't get your reward! Are you sure you want to close early?",
-                                                              VunglePlayAdOptionKeyIncentivizedAlertCloseButtonText: "Close",
-                                                              VunglePlayAdOptionKeyIncentivizedAlertContinueButtonText: "Keep Watching",
-                                                              VunglePlayAdOptionKeyIncentivizedAlertTitleText: "Careful!"])
+        let options = [VunglePlayAdOptionKeyUser: "test_user_id",
+                       VunglePlayAdOptionKeyIncentivizedAlertBodyText: "If the video isn't completed you won't get your reward! Are you sure you want to close early?",
+                       VunglePlayAdOptionKeyIncentivizedAlertCloseButtonText: "Close",
+                       VunglePlayAdOptionKeyIncentivizedAlertContinueButtonText: "Keep Watching",
+                       VunglePlayAdOptionKeyIncentivizedAlertTitleText: "Careful!"]
         do {
-            try self.sdk.playAd(self, options: (options as! [AnyHashable : Any]), placementID: kVungleTestPlacementID03)
+            try sdk.playAd(self, options: options, placementID: kVungleTestPlacementID03)
         }
         catch let error as NSError {
             print("Error encountered playing ad: + \(error)");
         }
     }
+}
+
+extension UIButton {
+    func updateState(enabled: Bool) {
+        isEnabled = enabled
+        alpha = enabled ? 1.0 : 0.45
+    }
+}
+extension ViewController: VungleSDKDelegate {
+    // MARK: - VungleSDKDelegate Methods
+    func vungleAdPlayabilityUpdate(_ isAdPlayable: Bool, placementID: String?) {
+        if (isAdPlayable) {
+            print("-->> Delegate Callback: vungleAdPlayabilityUpdate: Ad is available for Placement ID: \(String(describing: placementID))");
+        }
+        else {
+            print("-->> Delegate Callback: vungleAdPlayabilityUpdate: Ad is NOT availablefor Placement ID: \(String(describing: placementID))");
+        }
+        
+        if (placementID == kVungleTestPlacementID01) {
+            playButton1.updateState(enabled: isAdPlayable)
+        }
+        else if (placementID == kVungleTestPlacementID02) {
+            playButton2.updateState(enabled: isAdPlayable)
+            loadButton2.updateState(enabled: !isAdPlayable)
+        }
+        else if (placementID == kVungleTestPlacementID03) {
+            playButton3.updateState(enabled: isAdPlayable)
+            loadButton3.updateState(enabled: !isAdPlayable)
```

Comment:
I did change the format here, I created a method called `updateState` on `UIButton` to help clarify these points here.

Reply:
looks much better, thanks!